tasks/init.rb

```diff
@@ -5,7 +5,8 @@
 require 'open3'
 
 def get(fact)
-  cmd_string = "facter -p #{fact}"
+  facter = (%r{mingw} =~ RUBY_PLATFORM) ? '"C:\Program Files\Puppet Labs\Puppet\bin\facter"' : '/opt/puppetlabs/puppet/bin/facter'
```

Comment:
I'd lean towards PATH lookup on Windows. Not sure if we still support installing 32-bit on a 64-bit platform, but in that case this path would be wrong.

Reply:
That sounds like a better solution, but I'm not seeing the path to the Puppet executable directory in the environment variables within the environment of a running task.  When I dump all environment variables in a running task, I only see the base PATH:

```
    _ : /opt/puppetlabs/puppet/bin/pxp-agent
    PWD : /
    LANG : en_US.UTF-8
    PATH : /sbin:/usr/sbin:/bin:/usr/bin
    TERM : vt100
    SHLVL : 2
```

Is there a function I would use to extract the path from another location?

Reply:
Only true on Windows.

Reply:
On Windows, `environment.bat` will result in PATH including the Puppet/bin install location. On every other platform, you need to use a full path.